### ISSUE TITLE
tutorial_shallownet_prover.rs has an error

### ISSUE THREAD

**Issue by amyzx** [UCSB-TDS/ZEN#4]
Hi,
     I'm a student who wants to use zksnark to verify the correctness of neural network's computation. And I'm trying to run your tutorial code on windows 10 64-bit. 
     I have installed rust 1.47.0 and python 3.8.5, and the python files Mnist_end_to_end.py and Mnist_end_to_end_quant.py  work well. But I got the following error when running tutorial_shallownet_prover.rs.
    I guess it's an overflow error in the following line but have no idea how to fix it. 
![image](https://user-images.githubusercontent.com/8285571/117935557-6cce5700-b336-11eb-9e72-8ed5ab597a97.png)

![image](https://user-images.githubusercontent.com/8285571/116526970-bc5f5c80-a90c-11eb-8ce9-093d82bea822.png)

![image](https://user-images.githubusercontent.com/8285571/116527120-ed3f9180-a90c-11eb-89e5-6ce599defe5f.png)

Besides, I find that you have updated the code, but the document files are not updates. In your latest version, I cannot find tutorial_shallownet_prover.rs.  
Looking forward to your reply! Thanks!

**Comment by stechu** [comment]
@brucechin Can you reproduce this problem?

**Comment by amyzx** [comment]
Hi, I have solved this problem. I found that in zk-ml/src/mul_circuit.rs the following sentence has an error.
`if self.x.len() < 2u32.pow(SIMD_BOTTLENECK as u32) as usize {`
Because SIMD_BOTTLENECK is 210, 2^210 is a really large number and causes the overflow problem. I changed this line and tutorial_shallownet_prover.rs can run successfully now.
Besides, I have another question about tutorial_shallownet_verifier.rs.
It takes a long time to verify the proof, e.g. over 20 minutes to read crs.data.
I am not sure whether it's practical or acceptable for the user who wants to verify a neural network.
Looking forward to your reply. Thanks!
